core/src/main/java/uk/ac/bbsrc/tgac/miso/core/manager/UserAuthMisoRequestManager.java

```diff
@@ -2615,4 +2615,51 @@ public Long countRuns() throws IOException {
   public Long countRunsBySearch(String querystr) throws IOException {
     return backingManager.countRunsBySearch(querystr);
   }
+
+  @Override
+  public Run getLatestRunBySequencerPartitionContainerId(Long containerId) throws IOException {
+    Run o = backingManager.getLatestRunBySequencerPartitionContainerId(containerId);
+    if (readCheck(o))
+      return o;
+    else
+      throw new IOException("User " + getCurrentUser().getFullName() + " cannot read Run " + o.getId());
+  }
+
+  @Override
+  public Long countContainers() throws IOException {
+    return backingManager.countContainers();
+  }
+
+  @Override
+  public List<SequencerPartitionContainer<SequencerPoolPartition>> getContainersByPageSizeSearch(int offset, int limit, String querystr,
+      String sortDir, String sortCol) throws IOException {
+    User user = getCurrentUser();
+    List<SequencerPartitionContainer<SequencerPoolPartition>> accessibles = new ArrayList<>();
+    for (SequencerPartitionContainer<SequencerPoolPartition> spc : backingManager
+        .getContainersByPageSizeSearch(offset, limit, querystr, sortDir, sortCol)) {
+      if (spc.userCanRead(user)) {
+        accessibles.add(spc);
+      }
+    }
+    return accessibles;
+  }
+
+  @Override
+  public List<SequencerPartitionContainer<SequencerPoolPartition>> getContainersByPageAndSize(int offset, int limit, String sortDir,
+      String sortCol) throws IOException {
+    User user = getCurrentUser();
+    List<SequencerPartitionContainer<SequencerPoolPartition>> accessibles = new ArrayList<>();
+    for (SequencerPartitionContainer<SequencerPoolPartition> spc : backingManager
+        .getContainersByPageAndSize(offset, limit, sortDir, sortCol)) {
+      if (spc.userCanRead(user)) {
+        accessibles.add(spc);
+      }
+    }
+    return accessibles;
+  }
+
+  @Override
+  public Long countContainersBySearch(String querystr) throws IOException {
```

Comment:
Since this doesn't filter based on permissions, it may not match the number actually returned by the search query. I'm not sure what the solution is since looking up all of the containers would be slow